src/Metadata/Property/PropertyMetadataFactoryOptionsTrait.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Metadata\Property;
+
+use Symfony\Component\Serializer\Normalizer\AbstractNormalizer;
+
+trait PropertyMetadataFactoryOptionsTrait
+{
+    /**
+     * Gets a valid context for property metadata factories.
+     *
+     * @see https://github.com/symfony/symfony/blob/master/src/Symfony/Component/PropertyInfo/Extractor/SerializerExtractor.php
+     */
+    private function getFactoryOptions(array $context): array
+    {
+        $options = [];
+
+        if (isset($context[AbstractNormalizer::GROUPS])) {
+            $options['serializer_groups'] = $context[AbstractNormalizer::GROUPS];
+        }
+
+        if (isset($context['collection_operation_name'])) {
+            $options['collection_operation_name'] = $context['collection_operation_name'];
+        }
+
+        if (isset($context['item_operation_name'])) {
```

Comment:
It's could only be one of collection operation or item operation, so I think we could use `elseif` here?

Reply:
Yes, it was just a copy/paste from `AbstractItemNormalizer::getFactoryOptions()` but I can change it.

Reply:
I think we should also rename this method to something more descriptive: `getPropertyMetadataFactoryOptions`? But of course rename it in the `use` statement of `AbstractItemNormalizer` for BC.